// @canva/design/lib/cjs/sdk/design/public.js
"use strict"
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: Object.getOwnPropertyDescriptor(all, name).get
    });
}
_export(exports, {
    get addAudioTrack () {
        return addAudioTrack;
    },
    get addElementAtCursor () {
        return addElementAtCursor;
    },
    get addElementAtPoint () {
        return addElementAtPoint;
    },
    get addNativeElement () {
        return addNativeElement;
    },
    get bulkCreate () {
        return bulkCreate;
    },
    get createRichtextRange () {
        return createRichtextRange;
    },
    get editContent () {
        return editContent;
    },
    get getCurrentPageContext () {
        return getCurrentPageContext;
    },
    get getDefaultPageDimensions () {
        return getDefaultPageDimensions;
    },
    get getDesignMetadata () {
        return getDesignMetadata;
    },
    get getDesignTemplateMetadata () {
        return getDesignTemplateMetadata;
    },
    get getDesignToken () {
        return getDesignToken;
    },
    get initAppElement () {
        return initAppElement;
    },
    get openDesign () {
        return openDesign;
    },
    get overlay () {
        return overlay;
    },
    get publish () {
        return publish;
    },
    get requestExport () {
        return requestExport;
    },
    get resizeDesign () {
        return resizeDesign;
    },
    get searchResizePreset () {
        return searchResizePreset;
    },
    get selection () {
        return selection;
    },
    get setCurrentPageBackground () {
        return setCurrentPageBackground;
    },
    get ui () {
        return ui;
    }
});
const selection = canva_sdk.design.v2.designInteraction.selection;
const overlay = canva_sdk.design.v2.designInteraction.overlay;
const resizeDesign = canva_sdk.design.v2.designInteraction.resizeDesign;
const searchResizePreset = canva_sdk.design.v2.designInteraction.searchResizePreset;
const setCurrentPageBackground = canva_sdk.design.v2.designInteraction.setCurrentPageBackground;
const getDefaultPageDimensions = canva_sdk.design.v2.designInteraction.getDefaultPageDimensions;
const requestExport = canva_sdk.design.v2.export.requestExport;
const ui = canva_sdk.design.v2.dragAndDrop;
const addNativeElement = canva_sdk.design.v2.designInteraction.addNativeElement;
const addElementAtPoint = canva_sdk.design.v2.designInteraction.addElementAtPoint;
const addElementAtCursor = canva_sdk.design.v2.designInteraction.addElementAtCursor;
const addAudioTrack = canva_sdk.design.v2.designInteraction.addAudioTrack;
const getCurrentPageContext = canva_sdk.design.v2.designInteraction.getCurrentPageContext;
const initAppElement = canva_sdk.design.v2.designInteraction.initAppElement;
const getDesignToken = canva_sdk.design.v2.designInteraction.getDesignToken;
const getDesignMetadata = canva_sdk.design.v2.designInteraction.getDesignMetadata;
const getDesignTemplateMetadata = canva_sdk.design.v2.designInteraction.getDesignTemplateMetadata;
const createRichtextRange = canva_sdk.design.v2.designInteraction.createRichtextRange;
const editContent = canva_sdk.design.v2.designInteraction.editContent;
const openDesign = canva_sdk.design.v2.designInteraction.openDesign;
const publish = canva_sdk.design.v2.designInteraction.publish;
const bulkCreate = canva_sdk.design.v2.designInteraction.bulkCreate;
