// @canva/design/lib/esm/sdk/design/public.js
export const selection = canva_sdk.design.v2.designInteraction.selection;
export const overlay = canva_sdk.design.v2.designInteraction.overlay;
export const resizeDesign = canva_sdk.design.v2.designInteraction.resizeDesign;
export const searchResizePreset = canva_sdk.design.v2.designInteraction.searchResizePreset;
export const setCurrentPageBackground = canva_sdk.design.v2.designInteraction.setCurrentPageBackground;
export const getDefaultPageDimensions = canva_sdk.design.v2.designInteraction.getDefaultPageDimensions;
export const requestExport = canva_sdk.design.v2.export.requestExport;
export const ui = canva_sdk.design.v2.dragAndDrop;
export const addNativeElement = canva_sdk.design.v2.designInteraction.addNativeElement;
export const addElementAtPoint = canva_sdk.design.v2.designInteraction.addElementAtPoint;
export const addElementAtCursor = canva_sdk.design.v2.designInteraction.addElementAtCursor;
export const addAudioTrack = canva_sdk.design.v2.designInteraction.addAudioTrack;
export const getCurrentPageContext = canva_sdk.design.v2.designInteraction.getCurrentPageContext;
export const initAppElement = canva_sdk.design.v2.designInteraction.initAppElement;
export const getDesignToken = canva_sdk.design.v2.designInteraction.getDesignToken;
export const getDesignMetadata = canva_sdk.design.v2.designInteraction.getDesignMetadata;
export const getDesignTemplateMetadata = canva_sdk.design.v2.designInteraction.getDesignTemplateMetadata;
export const createRichtextRange = canva_sdk.design.v2.designInteraction.createRichtextRange;
export const editContent = canva_sdk.design.v2.designInteraction.editContent;
export const openDesign = canva_sdk.design.v2.designInteraction.openDesign;
export const publish = canva_sdk.design.v2.designInteraction.publish;
export const bulkCreate = canva_sdk.design.v2.designInteraction.bulkCreate;
