// @canva/design/lib/esm/sdk/design/fake/create_beta.js
import { createSyntheticDelay } from '../../utils/synthetic_delay';
import { FakeAutofillClient } from './fake_autofill_client';
import { FakeDesignInteractionClient } from './fake_design_interaction_client';
import { FakeDragAndDropClient } from './fake_drag_and_drop_client';
import { FakeExportClient } from './fake_export_client';
export function createBetaFakeDesignClients() {
    const syntheticDelay = createSyntheticDelay(10);
    const v2 = {
        designInteraction: new FakeBetaDesignInteractionClient(syntheticDelay),
        dragAndDrop: new FakeDragAndDropClient(syntheticDelay),
        export: new FakeExportClient(syntheticDelay),
        autofill: new FakeAutofillClient(syntheticDelay)
    };
    return {
        design: {
            v2
        }
    };
}
class FakeBetaDesignInteractionClient extends FakeDesignInteractionClient {
    async editContent(options, callback) {
        await this.delay();
        await callback({
            contents: [],
            sync: async ()=>{
                await this.delay();
            }
        });
    }
    getDesignMetadata() {
        return Promise.resolve({
            title: 'title',
            defaultPageDimensions: {
                width: 800,
                height: 600
            },
            pageMetadata: [
                {
                    title: 'title',
                    type: 'absolute',
                    pageDimensions: {
                        width: 800,
                        height: 600
                    }
                }
            ],
            durationInSeconds: 30
        });
    }
}
