// @canva/design/lib/cjs/sdk/design/fake/create_beta.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "createBetaFakeDesignClients", {
    enumerable: true,
    get: function() {
        return createBetaFakeDesignClients;
    }
});
const _synthetic_delay = require('../../utils/synthetic_delay');
const _fake_autofill_client = require("./fake_autofill_client");
const _fake_design_interaction_client = require("./fake_design_interaction_client");
const _fake_drag_and_drop_client = require("./fake_drag_and_drop_client");
const _fake_export_client = require("./fake_export_client");
function createBetaFakeDesignClients() {
    const syntheticDelay = (0, _synthetic_delay.createSyntheticDelay)(10);
    const v2 = {
        designInteraction: new FakeBetaDesignInteractionClient(syntheticDelay),
        dragAndDrop: new _fake_drag_and_drop_client.FakeDragAndDropClient(syntheticDelay),
        export: new _fake_export_client.FakeExportClient(syntheticDelay),
        autofill: new _fake_autofill_client.FakeAutofillClient(syntheticDelay)
    };
    return {
        design: {
            v2
        }
    };
}
class FakeBetaDesignInteractionClient extends _fake_design_interaction_client.FakeDesignInteractionClient {
    async editContent(options, callback) {
        await this.delay();
        await callback({
            contents: [],
            sync: async ()=>{
                await this.delay();
            }
        });
    }
    getDesignMetadata() {
        return Promise.resolve({
            title: 'title',
            defaultPageDimensions: {
                width: 800,
                height: 600
            },
            pageMetadata: [
                {
                    title: 'title',
                    type: 'absolute',
                    pageDimensions: {
                        width: 800,
                        height: 600
                    }
                }
            ],
            durationInSeconds: 30
        });
    }
}
